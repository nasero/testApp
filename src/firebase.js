import firebase from 'firebase/app';
import 'firebase/firestore';
import 'firebase/auth';

const firebaseConfig = {
    apiKey: 'AIzaSyBQ1cYtMlmidcIoRf3FxEzrO87KW17G3wM',
    authDomain: 'testapp-f1ef6.firebaseapp.com',
    projectId: 'testapp-f1ef6',
    storageBucket: 'testapp-f1ef6.appspot.com',
    messagingSenderId: '103885774192',
    appId: '1:103885774192:web:56838029614ac54b1b83b5'
};

firebase.initializeApp(firebaseConfig);

// utils
const db = firebase.firestore();
const auth = firebase.auth();

export { db, auth };
