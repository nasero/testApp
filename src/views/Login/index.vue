<!--testApp-Login: Vista login -->
<script src="./script.js" />
<style src="./style.sass" lang="sass" />
<template src="./template.pug" lang="pug" />
