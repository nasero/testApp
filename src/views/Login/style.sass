.login
    align-items: center,
    display: flex,
    flex-direction: column
    &__icon
        height: 4em,
        width: 4em
    &__form
        margin-top: 2em,
        width: 13em
    &__form-button
        margin-top: 1.5em
    &__error
        color: #ff6868,
        font-size: 0.75em,
        margin: 0,
        padding: 0