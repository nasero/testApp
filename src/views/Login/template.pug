.login.view
    img.login__icon(src="../../assets/padlock.svg")
    form.login__form(novalidate, @submit.prevent="")
        testApp-input(
            placeholder="Email",
            type="email",
            validations="email|required",
            name="email",
            label="E-mail"
            id="login-email",
            @testApp-input-value="setValue($event, 'email')")
        testApp-input(
            placeholder="password",
            type="password",
            validations="required",
            name="password",
            label="Password"
            id="login-pass",
            @testApp-input-value="setValue($event, 'password')")
        p.login__error(v-if="loginError") {{ loginError }}
        testApp-button.testApp-button--dark.login__form-button(type="submit" text="Log In", max-width, @testApp-button-clicked="login" :disableButton="disableButton")
