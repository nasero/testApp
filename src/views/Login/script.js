/**
 * Vista Login
 */

import { auth } from '../../firebase';
import testAppInput from '@/components/testApp-input';
export default {
    name: 'Login',
    components: {
        testAppInput
    },
    data () {
        return {
            email: '',
            password: '',
            lastSignIn: 0,
            loginError: ''
        };
    },
    computed: {
        disableButton () {
            return this.email === '' || this.password === '' || !!this.errors.items.length;
        }
    },
    methods: {
        /* istanbul ignore next */
        async login () {
            try {
                await auth.signInWithEmailAndPassword(this.email, this.password);
                this.$router.push('home');
            } catch (error) {
                this.loginError = error.message;
            }
        },
        setValue (payload, dataType) {
            this[dataType] = payload;
        }
    }
};
