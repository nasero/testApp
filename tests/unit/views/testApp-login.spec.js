import { createLocalVue, shallowMount } from '@vue/test-utils';
import Login from '@/views/Login';
import TestAppButton from '@/components/testApp-button';
import TestAppInput from '@/components/testApp-input';
import * as admin from '@/firebase';

admin.auth = {
    currentUser: {
        uid: '123'
    }
};

const localVue = createLocalVue();

localVue.component('testApp-button', TestAppButton);
localVue.component('testApp-input', TestAppInput);

describe('Home - Snapshot', () => {
    it('Snapshot', () => {
        const wrapper = shallowMount(Login, {
            localVue
        });
        expect(wrapper.element).toMatchSnapshot();
    });
});

describe('Login - computed', () => {
    it('disableButton ', () => {
        const wrapper = shallowMount(Login, {
            localVue
        });

        wrapper.vm.email = '123456';
        wrapper.vm.password = '123456';
        wrapper.vm.errors = {
            items: []
        };

        expect(wrapper.vm.disableButton).toBeFalsy();
    });
});

describe('Login - methods', () => {
    it('setValue', () => {
        const wrapper = shallowMount(Login, {
            localVue
        });

        wrapper.vm.setValue('foo@mail.com', 'email');

        expect(wrapper.vm.email).toBe('foo@mail.com');
    });
});
