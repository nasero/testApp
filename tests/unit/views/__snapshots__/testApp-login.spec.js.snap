// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home - Snapshot Snapshot 1`] = `
<div
  class="login view"
>
  <img
    class="login__icon"
    src="../../assets/padlock.svg"
  />
  <form
    class="login__form"
    novalidate="novalidate"
  >
    <testapp-input-stub
      id="login-email"
      label="E-mail"
      name="email"
      placeholder="Email"
      type="email"
      validations="email|required"
    />
    <testapp-input-stub
      id="login-pass"
      label="Password"
      name="password"
      placeholder="password"
      type="password"
      validations="required"
    />
    <!---->
    <testapp-button-stub
      class="testApp-button--dark login__form-button"
      disablebutton="true"
      maxwidth="true"
      text="Log In"
      type="submit"
    />
  </form>
</div>
`;
